android/java/org/chromium/chrome/browser/tab/BraveTab.java

```diff
@@ -0,0 +1,79 @@
+
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.tab;
+
+import android.content.SharedPreferences;
+import java.util.Calendar;
+import android.os.Build;
+
+import androidx.annotation.Nullable;
+
+import org.chromium.base.ContextUtils;
+import org.chromium.chrome.browser.tab.Tab;
+import org.chromium.chrome.browser.tabmodel.TabLaunchType;
+import org.chromium.chrome.browser.preferences.BackgroundImagesPreferences;
+import org.chromium.chrome.browser.ntp.sponsored.BackgroundImage;
+import org.chromium.chrome.browser.ntp.sponsored.SponsoredImage;
+import org.chromium.chrome.browser.ntp.sponsored.SponsoredImageUtil;
+
+public class BraveTab extends Tab {
```

Comment:
there are existing methods for attaching additional data to tabs/webcontents, but I'm not understanding why this code needs to be attached to the Tab at all vs being part of BraveNewTabPageView.java
There is nothing tab-specific about this and it seems very odd to get an image for every single tab when only the NTP will ever display it